test/custom_docstring_types_test.clj

```diff
@@ -0,0 +1,72 @@
+(ns custom-docstring-types-test
+  (:require [burpless :refer [docstring-type parameter-type run-cucumber step]]
+            [burpless.conversions :as conversions]
+            [camel-snake-kebab.core :as csk]
+            [clojure.data.json :as json]
+            [clojure.string :as str]
+            [clojure.test :refer [deftest is]])
+  (:import (clojure.lang IObj Keyword)
+           (io.cucumber.datatable DataTable)
+           (java.lang.reflect Type)))
+
+(def from-edn-steps
+  [(step :Given "my state starts out as an empty map"
+         (constantly {}))
+
+   (step :Given "I want to collect pairs of high and low temperatures under the {word} state key"
+         (fn [state keyword-name]
+           (assoc state :target-keyword (keyword (str/replace keyword-name #":" "")))))
+
+   (step :When "I have a table of the following high and low temperatures:"
+         ^:datatable
+         (fn [state ^DataTable dataTable]
+           (assoc state :highs-and-lows (.asLists dataTable ^Type Long))))
+
+   (step :Given "I have a key-value table:"
+         ^:datatable
+         (fn [_state ^DataTable dataTable]
+           (conversions/key-value-table->map dataTable)))
+
+
+   (step :Given "I have a table of data with key names in the first row:"
+         ^:datatable
+         (fn [_state ^DataTable dataTable]
+           (conversions/data-table->maps dataTable)))
+
+   (step :Then "my state should be equal to the following Clojure edn literal:"
+         ^{:docstring IObj}
```

Comment:
I didn't realize that `IObj` is edn. If you don't add this, it will just come in as a string correct?

Reply:
- If you omit the `^{:docstring IObj} metadata altogether, you'll get an arity mismatch exception:
```  Then my state should be equal to the following Clojure edn literal:                          # custom_docstring_types_test.clj:36
      io.cucumber.core.exception.CucumberException: Step [my state should be equal to the following Clojure edn literal:] is defined with 0 parameters at 'custom_docstring_types_test.clj:36'.
However, the gherkin step has 1 arguments:
 * DocString:
      """edn
      {:target-keyword :highs-and-lows
       :highs-and-lows [[81 49] [88 54] [76 56] [70 48] [81 55]]}
      """
Step text: my state should be equal to the following Clojure edn literal:
	at io.cucumber.core.runner.PickleStepDefinitionMatch.arityMismatch(PickleStepDefinitionMatch.java:83)
        ... (many more stack frames follow)
```

- If you leave it as just `^:docstring`, you'll get the `DocString` instance that wraps the raw string:
![image](https://github.com/danielmiladinov/burpless/assets/927811/05e78306-c36c-4437-afbf-27c181db2049)